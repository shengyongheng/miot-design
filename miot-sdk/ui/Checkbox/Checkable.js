import Checkable from 'mhui-rn/dist/components/checkbox/Checkable';
export default Checkable;