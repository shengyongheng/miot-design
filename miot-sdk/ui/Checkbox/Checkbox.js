import { Checkbox } from 'mhui-rn';
export default Checkbox;