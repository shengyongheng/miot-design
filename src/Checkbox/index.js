import { Checkbox } from 'mhui-rn-fixed';

export default Checkbox;
