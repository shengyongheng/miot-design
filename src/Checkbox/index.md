---
title: Checkbox
order: 0
group:
  path: '/basic'
  title: 基础组件
  order: 2
nav:
  title: '组件'
  path: /components
  order: 0
---

# Checkbox 复选框

## 简介

Checkbox 组件是一个基础的交互元素，用于触发操作。

## 用法

<!-- <code src="./index.tsx"></code> -->

```tsx
import React, { useState } from 'react';
import { Text } from 'react-native';
// @ts-ignore
import Checkbox from 'miot/ui/Checkbox';

const Index = () => {
  const [checked, setChecked] = useState(false);
  const [disabled] = useState(true);

  return (
    <>
      <Text>方形</Text>
      <Checkbox
        checked={checked}
        checkedColor="lightgreen"
        onValueChange={(checked) => setChecked(checked)}
      />
      <Text>圆形</Text>
      <Checkbox
        checked={checked}
        style={{ borderRadius: 30 }}
        checkedColor="lightgreen"
        onValueChange={(checked) => {
          console.log('checked:', checked);
          setChecked(checked);
        }}
      />
      <Text>禁用</Text>
      <Checkbox
        style={{ borderRadius: 30 }}
        disabled={disabled}
        checked={checked}
        checkedColor="lightgreen"
        onValueChange={(checked: boolean) => setChecked(checked)}
      />
    </>
  );
};

export default Index;
```

## API

| 属性  | 类型    | 默认值    | 说明                                           |
| ----- | ------- | --------- | ---------------------------------------------- |
| type  | string  | 'default' | 按钮类型，可选值为 'primary', 'dashed', 'link' |
| size  | string  | 'middle'  | 按钮尺寸，可选值为 'large', 'middle', 'small'  |
| shape | string  | 'default' | 按钮形状，可选值为 'circle', 'round'           |
| value | boolean | false     | 指定当前是否选中                               |
| ...   | ...     | ...       | ...                                            |
