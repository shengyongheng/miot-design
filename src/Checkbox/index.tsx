import React, { useState } from 'react';
import { Text } from 'react-native';
// @ts-ignore
import Checkbox from 'miot/ui/Checkbox';

const Index: React.FC = () => {
  const [checked, setChecked] = useState(false);
  return (
    <>
      <Text>// 方形</Text>
      <Checkbox
        style={{ width: 60, height: 60 }}
        checked={checked}
        checkedColor="lightgreen"
        onValueChange={(checked: boolean) => setChecked(checked)}
      />
      <Text>// 圆形</Text>
      <Checkbox
        style={{ width: 60, height: 60, borderRadius: 30 }}
        checked={checked}
        checkedColor="lightgreen"
        onValueChange={(checked: boolean) => setChecked(checked)}
      />
    </>
  );
};

export default Index;
