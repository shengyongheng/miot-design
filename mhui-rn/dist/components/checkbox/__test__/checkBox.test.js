
import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Checkbox from '../Checkbox';

configure({ adapter: new Adapter() });

describe('Checkbox', () => {
  describe('render', () => {
    it('render Checkbox', () => {
      const tree = renderer.create(
        <Checkbox checked={false} onValueChange={(checked) => { console.log(checked); }} />,
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('Interaction', () => {
    describe('onValueChange', () => {
      const mockOnPress = jest.fn();
      let instance1;
      beforeEach(() => {
        instance1 = shallow(<Checkbox
          value={false}
          onValueChange={mockOnPress}
        />).instance();
        jest.clearAllMocks();
      });
      it('should call onValueChange', () => {
        instance1.onValueChange();
        expect(mockOnPress).toHaveBeenCalled();
        expect(mockOnPress).toHaveBeenCalledTimes(1);
      });
      it('should toggle state', () => {
        instance1.onValueChange();
        expect(mockOnPress.mock.calls[0][0]).toBe(true);
        expect(instance1.state.checked).toBe(true);
      });
    });
  });
});
