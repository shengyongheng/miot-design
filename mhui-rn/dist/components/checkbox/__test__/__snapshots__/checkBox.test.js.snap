// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox render render Checkbox 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgba(240, 240, 240, 1)",
      "borderColor": "rgba(0,0,0,0.1)",
      "borderWidth": 1,
      "height": 40,
      "width": 40,
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View />
  </View>
</View>
`;
